sonic_platform_base/sonic_xcvr/api/public/cmis.py

```diff
@@ -520,16 +522,18 @@ def get_rx_power(self):
         '''
         This function returns RX input power in mW on each media lane
         '''
-        rx_power_support = self.get_rx_power_support()
-        if rx_power_support is None:
-            return None
-
         rx_power = ["N/A" for _ in range(self.NUM_CHANNELS)]
 
+        rx_power_support = self.get_rx_power_support()
+        if not rx_power_support:
+            return rx_power
+
         if rx_power_support:
             rx_power = self.xcvr_eeprom.read(consts.RX_POWER_FIELD)
             if rx_power is not None:
                 rx_power = [rx_power['OpticalPowerRx%dField' %i] for i in range(1, self.NUM_CHANNELS+1)]
+            else:
+                return None
```

Comment:
@stephenxs  we don't need this else? `rx_power=None` already at line 534 so the return at line 538 will indeed return None when 'rx_power' = None

Reply:
Updated